src/typing/type_inference_js.ml

```diff
@@ -1292,8 +1292,29 @@ and toplevels cx stmts =
       if uc < List.length stmts
       then (
         let msg = "unreachable code" in
-        let (loc, _) = List.nth stmts uc in
-        Flow_js.add_warning cx [mk_reason "" loc, msg]
+        let rec drop n lst =
+          match (n, lst) with
+          | (_, []) -> []
+          | (0, l) -> l
+          | (x, h :: t) -> drop (x-1) t
+        in
+        let trailing = drop uc stmts in
+        trailing |> List.iter Ast.Statement.(fun stmt ->
+          match stmt with
+          (* function declarations are hoisted, so not unreachable *)
+          | (_, FunctionDeclaration {
```

Comment:
@Eyyub thanks, I had played around trying to figure out how to do that, but didn't quite hit the right syntax, so I copied this pattern from elsewhere.

Can you point me at some reading on pattern matching with modules/records? I still want to match a `VariableDeclaration` with a guard that checks that for all its `declarations`, `init` is `None`, but I think I need to learn a bit more to get that working without type errors.